Lab20.md

```diff
@@ -14,38 +15,125 @@ wget https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kuberne
+
+* PersistentVolumeClaim (wp-pv-claim)
+* volumeMounts (wordpress-persistent-storage)
+```
+more wordpress.yaml
+```
+
+How much storage was allocated for Wordpress?
+What type of storage was allocated?
+Where in the container in the storage mounted?
+What does "accessModes: ReadWriteOnce" mean?
+
+### MySQL Description
+
+Examine the MySQL application descriptions. Take note of:
+
+* PersistentVolumeClaim (mysql-pv-claim)
+* volumeMounts (mysql-persistent-storage)
+
+```
+more mysql.yaml
+```
+
+How much storage was allocated for Wordpress?
+What type of storage was allocated?
+Why is this claim labeled with "app: wordpress"?
+
 
 ## Create Applications
 
+Go ahead and start up the applications which in turn will also allocate the storage through Rook.
 ```
 kubectl create -f mysql.yaml
 kubectl create -f wordpress.yaml
 ```
 
-Then let's wait for the pods to be ready
+## Volume Claims
+
+The PVCs (Persistent Volume Claims) will be allocated as part of this deployment. Watch as Rook allocates these storage resources.
+
 ```
-watch -n1 -d kubectl get pods
+watch -n1 -d kubectl get PersistentVolumeClaim
 ```
-ctrl-c once the pods are `Running`
+
+Once the storage is allocated, "ctrl-c" out of this watch and verify that the rest of the deployment has started.
 
 ## Examine the created resources
 
+Validate that all the Wordpress and MySQL pods are running.
+
+```
+watch -n1 -d kubectl get pods
+```
+
+Once all the pods are started, "ctrl-c" out of this watch and verify that the full deployment and services are running OK.
+
 ```
 kubectl get Deployments
-kubectl get PersistentVolumeClaim
 kubectl get Services
 ```
 
-Run `rbd ls -p replicapool` in the toolbox container
+## Examine Ceph setup via Toolbox
+
+Let's take a deeper look at has been setup in Ceph. Pull up the Ceph Toolbox container and run through some Ceph commands to examine the configuration.
+
+```
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.0/cluster/examples/kubernetes/ceph/toolbox.yaml
+kubectl --namespace rook-ceph exec -it $(kubectl --namespace rook-ceph get pod -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') bash
+```
+
+List out all the created pools created in an earlier lab.
+```
+rados df
+```
+
+List the images associated with the pool "replicapool" listed above.
+```
+rbd -p replicapool ls -l
+```
+
+Get the details on each image (volume claim) from the listing above.
+```
+rbd -p replicapool info <YOUR_PVC_IMAGE_NAME>
+```
+
+Be sure to "exit" out of the toolbox and back to the node shell when continuing with the lab.
+
+## Examine Ceph setup via Dashboard
+
+Through the Ceph dashboard (web), examine the same items that were examined via the toolbox. This information is under the "Pool" and "Block" menu items.
+
+The following will get you the dashboard URL:
+```
+kubectl -n rook-ceph expose svc rook-ceph-mgr-dashboard --name ceph-dashboard-external --type NodePort
```

Comment:
`kubectl expose` is already done in Lab15, so doing it a second time will result in an error?